src/paths.ts

```diff
@@ -6,34 +6,95 @@ import * as path from 'path';
 import {decode} from 'urlencode';
 
 /** converts a server-side XDebug file URI to a local path for VS Code with respect to source root settings */
-export function convertDebuggerPathToClient(fileUri: string|url.Url, localSourceRoot?: string, serverSourceRoot?: string): string {
+export function convertDebuggerPathToClient(fileUri: string|url.Url, localSourceRoot?: string, serverSourceRoot?: string, pathMapping?: { [index: string]: string; }): string {
+
     if (typeof fileUri === 'string') {
         fileUri = url.parse(fileUri);
     }
+
     // convert the file URI to a path
     let serverPath = decode(fileUri.pathname!);
+
     // strip the trailing slash from Windows paths (indicated by a drive letter with a colon)
     const serverIsWindows = /^\/[a-zA-Z]:\//.test(serverPath);
+
     if (serverIsWindows) {
         serverPath = serverPath.substr(1);
     }
+
+    if ( typeof pathMapping !== 'undefined' ) {
+        let mappedServerSource: string;
+
+        for (mappedServerSource of Object.keys(pathMapping) ) {
+            let mappedLocalSource: string = pathMapping[mappedServerSource];
+
+            // normalize slashes for windows-to-unix
+            // I thought normalize did this automagically..
+            if ( process.platform !== 'win32' ) {
+                mappedServerSource = mappedServerSource.replace(/\\/g, '/');
```

Comment:
This handles the case where local is Linux and server is Windows. This should be the only time backslashes need be converted because local windows will normalize the backslashes in path.relative just fine. Linux just sees the backslashes as part of the filename so path.relative returns the wrong value.